libs/src/api/file.js

```diff
@@ -47,14 +50,17 @@ class File extends Base {
    * returning the response content.
    * @param {string} cid IPFS content identifier
    * @param {Array<string>} creatorNodeGateways fallback ipfs gateways from creator nodes
+   * @param {string?} filename optional filename for the download
+   * @param {boolean?} usePublicGateways
    */
-  async downloadCID (cid, creatorNodeGateways) {
-    const gateways = publicGateways
-      .concat(creatorNodeGateways)
-    const urls = gateways.map(gateway => `${gateway}${cid}`)
+  async downloadCID (cid, creatorNodeGateways, filename, usePublicGateways = false) {
+    const gateways = usePublicGateways ? publicGateways.concat(creatorNodeGateways) : creatorNodeGateways
+    const urls = gateways.map(gateway => `${gateway}${cid}?filename=${filename}`)
```

Comment:
i think we should use `path.join` here instead of simple string concat

Reply:
ya sg, ill do this